src/org/daisy/dotify/formatter/impl/volume/TocSequenceEventImpl.java

```diff
@@ -121,16 +123,23 @@ public BlockSequence getBlockSequence(FormatterContext context, DefaultContext v
 					getSequenceProperties());
 			fsm.appendGroup(getTocStart(vars));
 			if (getRange()==TocProperties.TocRange.VOLUME) {
-				Collection<Block> volumeToc = data.filter(refToVolume(vars.getCurrentVolume(), crh));
+                int currentVolume = vars.getCurrentVolume();
+                Collection<Block> resumedBlocks = data.filterEntryOnResumed(rangeToVolume(currentVolume, crh));
+				Collection<Block> volumeToc = data.filterEntry(refToVolume(currentVolume, crh));
+                if (resumedBlocks.isEmpty() && volumeToc.isEmpty()) {
+					return null;
```

Comment:
What happens if one of them is empty but not both?

Reply:
Then one of them (but not both) are appended to the TOC. See the code that immediately follows.